processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -806,7 +781,7 @@ public JavaExpressionMappingBuilder javaExpression(String javaExpression) {
         }
 
         public PropertyMapping build() {
-            Assignment assignment = new SourceRHS( javaExpression, null, existingVariableNames );
+            Assignment assignment = new SourceRHS( javaExpression, null, existingVariableNames, "" );
```

Comment:
Sorry I didn't express myself really well. In case  Why are we using an empty one here and not the `javaExpression`? In the `ConstantMappingBuilder` you have `String sourceErrorMessagePart = "constant '" + constantExpression + "'";` which is then used in the `SourceRHS`